po/R-zh_CN.po

```diff
@@ -87,87 +87,95 @@ msgid "; filled with NA"
 msgstr ""
 
 msgid "A column may not be called .SD. That has special meaning."
-msgstr ""
+msgstr "无法将列命名为 .SD，因为 .SD 为特殊符号。"
 
 msgid "class must be length 1"
-msgstr ""
+msgstr "class 的长度必须为 1"
 
 msgid "between has been passed an argument x of type logical"
-msgstr ""
+msgstr "传入 between 的参数 x 为逻辑（logical）型"
 
 msgid ""
 "'between' function the 'x' argument is a POSIX class while 'lower' was not, "
 "coercion to POSIX failed with:"
 msgstr ""
+"'between' 中的 'x' 参数为 POSIX 类，而 'lower' 并不是，"
+"将 'lower' 强制转换成 POSIX 失败："
 
 msgid ""
 "'between' function the 'x' argument is a POSIX class while 'upper' was not, "
 "coercion to POSIX failed with:"
 msgstr ""
+"'between' 中的 'x' 参数为 POSIX 类，而 'upper' 并不是，"
+"将 'upper' 强制转换成 POSIX 失败："
 
 msgid ""
 "'between' lower= and upper= are both POSIXct but have different tzone "
 "attributes:"
 msgstr ""
+"'between' 中的 lower= 和 upper= 均为 POSIXct 类型但却有不同的时区"
+"属性（tzone）："
 
 msgid ". Please align their time zones."
-msgstr ""
+msgstr "。请确保二者的时区一致。"
 
 msgid ""
 "'between' arguments are all POSIXct but have mismatched tzone attributes:"
 msgstr ""
+"'between' 的参数均为 POSIXct 类型但时区属性（tzone）不匹配："
 
 msgid ". The UTC times will be compared."
-msgstr ""
+msgstr "。将采用 UTC 时间进行比较。"
 
 msgid "trying to use integer64 class when 'bit64' package is not installed"
-msgstr ""
+msgstr "试图使用 intger64 类型但 'bit64' 包尚未安装"
 
 msgid ""
 "Not yet implemented NAbounds=TRUE for this non-numeric and non-character type"
 msgstr ""
+"将 NAbounds=TRUE 应用于非数值（numeric）型和非字符串（character）型的功能尚未开发"
 
 msgid "Some lower>upper for this non-numeric and non-character type"
-msgstr ""
+msgstr "对于该非数值（numeric）型和非字符串（character）型的输入，存在一部分下界（lower）> 上界（upper）的情况"
 
 msgid "RHS has length()"
-msgstr ""
+msgstr "右手侧（RHS）的长度为"
 
 msgid "; expecting length 2."
-msgstr ""
+msgstr "；其长度应为 2。"
 
 msgid "c"
 msgstr ""
 
 msgid "Perhaps you meant %s?"
-msgstr ""
+msgstr "或许你指的是 %s？"
 
 msgid "The first element should be the lower bound(s);"
-msgstr ""
+msgstr "第一个元素应为下界；"
 
 msgid "the second element should be the upper bound(s)."
-msgstr ""
+msgstr "第二个元素应为上界。"
 
 msgid "x."
 msgstr ""
 
 msgid "is type"
-msgstr ""
+msgstr "的类型为"
 
 msgid "which is not supported by data.table join"
-msgstr ""
+msgstr "，该类型无法用于 data.table 的联结"
```

Comment:
It's difficult to translate such a simple and common word `join`. I found the most common translation is 连接 which is not really best. 联结 is better, another option is 联接. Many Chinese documentations actually just use `join` since it's a very simple word.

In any case, I think we should include the original word with (join). As for which word to use, I vote for 联接, but 连接 is acceptable if it's already established (but it's the worst in 3 options).

Reply:
I'm quite surprised `join` nor `merg` really show up anywhere in the base translation files 😳 so we don't really have any canon to refer to within R...

FWIW, Chinese Wikipedia is actually using 连接 for SQL join:

https://zh.wikipedia.org/wiki/%E8%BF%9E%E6%8E%A5

Reply:
Yes 连接 is a more common usage. However it only have the meaning of `linking`, I would think it just means you `cbind` two things together and don't have the part of matching values, which both `联结` and `联接` provided.

Between `联结` and `联接`, `联结` is hinting a strong binding that hard to break, more like the ending state, `联接` have the process of connecting two things together so I think it's better.

Reply:
In that sense, maybe 连接 is closer to `join`, which also doesn't have any "intrinsic" sense of matching to me; `merge` gets closer to that.

That said, probably 联接 is preferred since it's more important for users to understand the meaning rather than to match the English. In English, I know we are implicitly bootstrapping on users' understanding of `join`s in the relational database sense which is well known. I leave it up to you to know how well-known the different usages are :)